src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-                            dataport = stoi(*j);
-                        } else {
-                            partitionIDs.push_back(*j);
-                            totalPartitions += 1;
-                        }
-                    }
-                    graphPartitionedHosts.insert(pair<string, JasmineGraphInstanceService::workerPartitions>(
-                        hostName, {port, dataport, partitionIDs}));
-                }
-                count++;
-            }
-            /*Receive file*/
-            write(connFd, JasmineGraphInstanceProtocol::SEND_FILE_NAME.c_str(),
-                  JasmineGraphInstanceProtocol::SEND_FILE_NAME.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::SEND_FILE_NAME, "info");
+        if (sockfd < 0) {
+            std::cout << "Cannot create socket" << std::endl;
+            return;
+        }
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string fileName = (data);
-            instance_logger.log("Received File name: " + fileName, "info");
-            write(connFd, JasmineGraphInstanceProtocol::SEND_FILE_LEN.c_str(),
-                  JasmineGraphInstanceProtocol::SEND_FILE_LEN.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::SEND_FILE_LEN, "info");
+        server = gethostbyname(host.c_str());
+        if (server == NULL) {
+            std::cout << "ERROR, no host named " << server << std::endl;
+            return;
+        }
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string size = (data);
-            instance_logger.log("Received file size in bytes: " + size, "info");
-            write(connFd, JasmineGraphInstanceProtocol::SEND_FILE_CONT.c_str(),
-                  JasmineGraphInstanceProtocol::SEND_FILE_CONT.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::SEND_FILE_CONT, "info");
+        bzero((char *)&serv_addr, sizeof(serv_addr));
+        serv_addr.sin_family = AF_INET;
+        bcopy((char *)server->h_addr, (char *)&serv_addr.sin_addr.s_addr, server->h_length);
+        serv_addr.sin_port = htons(port);
+        if (Utils::connect_wrapper(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
+            std::cout << "ERROR connecting" << std::endl;
+            // TODO::exit
+            return;
+        }
 
-            string fullFilePath =
-                Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder") + "/" + fileName;
-            int fileSize = atoi(size.c_str());
-            while (Utils::fileExists(fullFilePath) && Utils::getFileSize(fullFilePath) < fileSize) {
-                bzero(data, INSTANCE_DATA_LENGTH + 1);
-                read(connFd, data, INSTANCE_DATA_LENGTH);
-                line = (data);
+        bzero(data, 301);
```

Comment:
```suggestion
        bzero(data, INSTANCE_DATA_LENGTH + 1);
```

Reply:
`bzero()` is no longer used in the next PR to clear the `data` array. It's handled in the wrapper function without `bzero()`

Reply:
Ok